common/src/main/java/net/william278/husktowns/hook/Pl3xMapHook.java

```diff
@@ -0,0 +1,147 @@
+package net.william278.husktowns.hook;
+
+import net.pl3x.map.core.Pl3xMap;
+import net.pl3x.map.core.event.EventHandler;
+import net.pl3x.map.core.event.EventListener;
+import net.pl3x.map.core.event.server.Pl3xMapEnabledEvent;
+import net.pl3x.map.core.event.world.WorldLoadedEvent;
+import net.pl3x.map.core.event.world.WorldUnloadedEvent;
+import net.pl3x.map.core.markers.Point;
+import net.pl3x.map.core.markers.layer.SimpleLayer;
+import net.pl3x.map.core.markers.marker.Marker;
+import net.pl3x.map.core.markers.option.Options;
+import net.pl3x.map.core.markers.option.Tooltip;
+import net.william278.husktowns.HuskTowns;
+import net.william278.husktowns.claim.TownClaim;
+import net.william278.husktowns.claim.World;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+public class Pl3xMapHook extends MapHook implements EventListener {
+
+    private static final String CLAIMS_LAYER = "claim_markers";
+    private final ConcurrentLinkedQueue<TownClaim> claims = new ConcurrentLinkedQueue<>();
+
+    public Pl3xMapHook(@NotNull HuskTowns plugin) {
+        super(plugin, "Pl3xMap");
+    }
+
+    @Override
+    public void onEnable() {
+        Pl3xMap.api().getEventRegistry().register(this);
+
+        // TODO: Pl3xMap's enable and world load events are currently being ran before this method meaning that the claims layer isn't loading on server startup
+        if (Pl3xMap.api().isEnabled()) Pl3xMap.api().getWorldRegistry().forEach(this::registerLayers);
```

Comment:
This is totally fine -- as this is dependent on load order. But be sure to convert this to be a braced statements -- avoid stuff like this where possible generally

Reply:
Fixed in [4cb4c83](https://github.com/WiIIiam278/HuskTowns/pull/303/commits/4cb4c83d7605c2ddf4f3161c348c9529c479b7fb)

Is there a reason for avoiding non-braced statements? I actually didn't know about this and use them quite often in my code.

Reply:
> Is there a reason for avoiding non-braced statements? I actually didn't know about this and use them quite often in my code.

Not really -- just personal preference. I've found in the past using them can make code a bit harder to read and more often lead to bugs -- e.g.
```java
if (foo == true && bar == false)
   System.out.println("test")
   return true;
return false;
```
will _always_ misleadingly return `true`, as that second indented statement is not actually protected by the `if` statement&mdash;hence, braces help clarify. I do agree this argument is less strong in single-line cases, but I think consistency is important